src/internal/components/focus-portal/index.tsx

```diff
@@ -0,0 +1,132 @@
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+  const portalBeforeRef = useRef<HTMLDivElement | null>(null);
+  const portalAfterRef = useRef<HTMLDivElement | null>(null);
+
+  const focusBeforeProxy = () => {
+    const allTabbableElements = getTabbables(document.documentElement);
+    const beforeIndex = proxyBeforeRef.current ? allTabbableElements.indexOf(proxyBeforeRef.current) : -1;
+    if (beforeIndex !== -1 && beforeIndex !== 0) {
+      allTabbableElements[beforeIndex - 1].focus();
+    } else {
+      document.body.focus();
+    }
+  };
+
+  const focusInsideProxy = (backwards = false) => {
+    if (wrapperRef.current) {
+      const element = backwards ? getLastTabbable(wrapperRef.current) : getFirstTabbable(wrapperRef.current);
+      if (element) {
+        element.focus();
+        return true;
+      }
+      return false;
+    }
+  };
+
+  const focusStartofDOM = () => {
+    const element = getFirstTabbable(document.documentElement);
+    if (element) {
+      // Potential infinite loop if proxy is first or only tabbable element.
+      element.focus();
+      return true;
+    }
+    document.body.focus();
+    return true;
+  };
+
+  const focusAfterProxy = () => {
+    if (proxyAfterRef.current) {
+      const allTabbableElements = getTabbables(document.documentElement);
+      const anchorIndex = allTabbableElements.indexOf(proxyAfterRef.current);
+      if (anchorIndex !== -1 && anchorIndex !== allTabbableElements.length - 1) {
+        allTabbableElements[anchorIndex + 1].focus();
+        return true;
+      }
+    }
+    return focusStartofDOM();
+  };
+
+  const focusBeforePortal = () => {
+    if (portalBeforeRef.current) {
+      const allTabbableElements = getTabbables(document.documentElement);
+      const anchorIndex = allTabbableElements.indexOf(portalBeforeRef.current);
+      if (anchorIndex !== -1 && anchorIndex !== 0) {
+        allTabbableElements[anchorIndex - 1].focus();
+        return true;
+      }
+    }
+    return focusStartofDOM();
+  };
+
+  const focusAfterPortal = () => {
+    if (portalAfterRef.current) {
+      const allTabbableElements = getTabbables(document.documentElement);
+      const anchorIndex = allTabbableElements.indexOf(portalAfterRef.current);
+      if (anchorIndex !== -1 && anchorIndex !== allTabbableElements.length - 1) {
+        allTabbableElements[anchorIndex + 1].focus();
+        return true;
+      }
+    }
+    return focusStartofDOM();
+  };
+
+  return (
+    <>
+      <FocusDetector
+        ref={proxyBeforeRef}
+        disabled={!children}
+        onFocus={() => {
+          focusInsideProxy() || focusAfterProxy();
+        }}
+      />
+      <Portal container={container}>
+        <FocusDetector
+          ref={portalBeforeRef}
+          disabled={!children}
+          onFocus={event => {
+            if (wrapperRef.current?.contains(event.relatedTarget)) {
+              focusBeforeProxy();
+            } else {
+              focusAfterPortal();
```

Comment:
Why this is needed? I would imagine this focus to be only triggered with shift-tab from inside the portal.